### ISSUE TITLE
class_id always mismatch. persists in 1.6.4

### ISSUE THREAD

**Issue by golden452** [roboflow/rf-detr#961]
### Search before asking

- [x] I have searched the RF-DETR issues and found no similar bug report.

### Bug

for fine-tuned model, class_id always mismatch. 
predicted classes are always `real class id - 1`.  i.e. class 10 is predicted as 9, and 9 as 8, etc. 
I tested in a brand new env with RF-DETR v1.6.4 freshly installed and this still happens. 

_I have 10 kinds of objects and a 10-class dataset._ 
For fine tune training, I use COCO format annotations. there are 10 category/class ids, ranging from 1 to 10. Following discussions here in the GitHub Issues, I left 0: Background out.  only class_id 1 to 10 are defined.

```
categories	
0	{ id: 1, name: "1", supercategory: "None" }
1	{ id: 2, name: "2", supercategory: "None" }
2	{ id: 3, name: "3", supercategory: "None" }
3	{ id: 4, name: "4", supercategory: "None" }
4	{ id: 5, name: "5", supercategory: "None" }
5	{ id: 6, name: "6", supercategory: "None" }
6	{ id: 7, name: "7", supercategory: "None" }
7	{ id: 8, name: "8", supercategory: "None" }
8	{ id: 9, name: "9", supercategory: "None" }
9	{ id: 10, name: "10", supercategory: "None" }
```

And during training, I explicitly set `num_classes to 10`. 

```
from rfdetr import RFDETRLarge #,RFDETRMedium,RFDETRSmall,RFDETRBase
model = RFDETRLarge(num_classes = 10)

model.train(
    dataset_dir='datadir',
    epochs=10,
    batch_size=6,
    grad_accum_steps=2,
    lr=1e-6,
    output_dir='outputdir'
)

```

However, predicted class id is always 1 shifted to the left. No matter I initiate with num_classes = 10 or 11. 

```
from rfdetr import RFDETRLarge


model = RFDETRLarge(pretrain_weights = MODEL_PATH,num_classes = 11)

detections = model.predict('testimage', 
                           threshold=0.1)

print(model.model_config.num_classes)
print(detections.class_id)
```

```
11
[9 9 4 4 5 5]
```
What should be a class 10 is predicted as 9, 9 as 8, etc.

I considered just "hard" adding 1 to the prediction class id. But I worry if I got some underlying stuff completely wrong. 

I would sincerely appreciate if you could let me know where I did wrong and how to solve this issue. 

**Also a definitive guide on how to define number of classes and class id would also be greatly appreciated. Do we count in class 0 Background? What `number of classes` should be used when initiating models for training and predicting? How to define class ids in training annotation file? Discussions in Issues have conflicting answers.**

### Environment

RF-DETR 1.6.4

### Minimal Reproducible Example

```
model = RFDETRLarge(pretrain_weights = MODEL_PATH,num_classes = 11)

detections = model.predict('testimage', 
                           threshold=0.1)
```

### Are you willing to submit a PR?

- [ ] Yes, I'd like to help by submitting a PR!

**Comment by zhaoshuo1223** [comment]
The parameter num_classesonly affects the configuration of the head. Regarding the issue where the number of categories is always -1, you can perform a global search for the parameter: remap_category_ids, and add a breakpoint to debug and examine the specific implementation.

**Comment by golden452** [comment]
> The parameter num_classesonly affects the configuration of the head. Regarding the issue where the number of categories is always -1, you can perform a global search for the parameter: remap_category_ids, and add a breakpoint to debug and examine the specific implementation.

Thanks for your input. Still got some problems. 

I tested with a simpler digit dataset with only 2 classes, hand written number digit 1 an 2. 
the training COCO annotation is 1-based. so `class_id` is 1 for 1 and 2 for 2.  `class_name` are strings "1" and "2" for class_id 1 and 2.

**I see that model prediction results are mostly correct,** model has 2 classes. Only class_id now becomes 0-based -- 1 are labeled 0 and 2 are labeled 1. Is this the expected behavior?

**But the exported .onnx weights is still confusing.** -- it has 3 classes. Does it include Background class? if so, which one is?


with the pth weight, 
```
print(model.model_config.num_classes)
```
shows model has `2` classes, which is correct. 

```
detections = model.predict('/home/user/Desktop/catdog/images/000000407083.jpg', #2
                           threshold=0.1)
detections
```
And by checking `detections`, I see class_name are correctly mapped. Only the numerical `class_id` becomes 0-based during training. 
```
class_id=array([0, 0, 0]),'class_name': array(['1', '1', '1'] #this is a result for an image of digit 1.
```

**But checking .onnx weights exported by the official code, I see that output now has 3 classes. One more than what should be.** 
```
  - Name: labels
    Shape: [1, 300, 3]
```
A visual check shows that most label 0 are digit 1, which is correct if the result is 0 based. Then, what is class 2 (the third class as it's 0-based) in this case?

I can't find official documentations on this issue. Class numbering conventions, include or not Background, 0 or 1-based, which one is Background for logits output, etc. These questions have made prediction results very confusing. A confirmation on label behaviors from developers would be greatly appreciated.

**Comment by zhaoshuo1223** [comment]
Global search parameter, remap_category_ids: The labels actually learned during model training are remapped to [0, 1, ...] (to address the issue of discontinuous class_id in annotation files).
Global search: num_classes + 1, actually when initializing the classification head, +1 is added, i.e., the last category is treated as the background category (this is written as a comment in a certain line).

**Comment by zhaoshuo1223** [comment]
I think a documentation can be added here.

**Comment by golden452** [comment]
Thank you very much. 

`model.model_config.num_classes` returning `2` classes while onnx label having 3 classes  `Shape: [1, 300, 3]` confuses me a bit, so I asked just to be sure. Now I know remapping to 0-based happens during training, and the last category is added as the background. 

A documentation would help clarify. And I don't have other questions on this topic so I'm closing the issue. 

Thanks again. your answer really helped!
